src/devices/controller.c

```diff
@@ -208,8 +208,7 @@ int32_t controller_set_text(controller_id_e_t id, uint8_t line, uint8_t col, con
 		col++;
 
 	char* buf = (char*)malloc(CONTROLLER_MAX_COLS + 1);
-	strcpy(buf, str);
-	buf[CONTROLLER_MAX_COLS] = '\0';
+	strlcpy(buf, str, CONTROLLER_MAX_COLS + 1);
```

Comment:
I have a feeling this won't compile. It's a nonstandard function and I'm skeptical that it is provided in our libc implementation.

That said, there shouldn't be anything wrong with just using `strncpy` here

Reply:
That's reasonable, it's an odd one. The only real benefit of `strlcpy()` over `strncpy()` is the automatic addition of the null terminator, which could just be added manually after a `strncpy()` call if the current code doesn't compile.

Reply:
`char* buf = strncpy(str, CONTROLLER_MAX_COLS + 1);` would do the whole thing in one line.